Makefile

```diff
@@ -23,10 +29,8 @@ install-tools:
 	@echo ">> fetching dep"
 	@go get -u github.com/golang/dep/cmd/dep
 
-proto:
+proto: deps
```

Comment:
dep `requires` is totally fine,  but I would not put `deps` as a command dependency. I am doing `make proto` again, and I would like to avoid unnecessary `make dep` (: I think we can explicitly call these methods just fine. What do you think? Makefile has some caching possibilities, but I am not familiar how to take advantage of these.